src/System.Net.Primitives/src/System.Net.Primitives.csproj

```diff
@@ -124,19 +140,51 @@
     <Compile Include="$(CommonPath)\Interop\Windows\NtDll\Interop.RtlIpv6StringToAddressEx.cs" >
       <Link>Common\Interop\Windows\NtDll\Interop.RtlIpv6StringToAddressEx.cs</Link>
     </Compile>
-    <Compile Include="$(CommonPath)\Interop\Windows\SChannel\Interop.SchProtocols.cs" >
-      <Link>Common\Interop\Windows\SChannel\Interop.SchProtocols.cs</Link>
+  </ItemGroup>
+  
+  <ItemGroup Condition=" '$(TargetsUnix)' == 'true' ">
+    <Compile Include="System\Net\IPAddressPal.Unix.cs" />
```

Comment:
The result is effectively the same: IPAddress.Unix.cs would look essentially the same as IPAddressPal.Unix.cs. The call, I think, is one of tradeoffs: if the code that implements a particular type is going to be mostly platform-specific, then it makes sense to add the platform-specific code to that type directly and PAL things out with a partial class. If, on the other hand, the code that implements a type is mostly platform-agnostic (or if the code that sits behind the PAL has little to no dependency on the internals of the type), then it makes sense to split the PAL out into its own type. Personally, I prefer the latter when possible: I think  that it makes sense to separate concerns in the fashion presented.